### PR TITLE
[GAUAlpha P0] PretrainedConfig and model unit test

### PR DESCRIPTION
**Pull request by w5688414**
### PR types

+Function optimization

### PR changes

+ APIs

### CODE REVIEW

File: paddlenlp/transformers/gau_alpha/configuration.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    >>> from paddlenlp.transformers import GAUAlphaModel, GAUAlphaConfig
+    >>> # Initializing a GAUAlpha chinese_GAU-alpha-char_L-24_H-768style configuration
+    >>> configuration = GAUAlphaConfig()
+    >>> # Initializing a model from the  style configuration
+    >>> model = GAUAlphaModel(configuration)
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "gau_alpha"
+    attribute_map: Dict[str, str] = {"dropout": "classifier_dropout", "num_classes": "num_labels"}
```
Comment (wj-Mcat) [inline paddlenlp/transformers/gau_alpha/configuration.py]: 如果这个模型在 example 中根本就没有示例，所以dropout、和 num_classes的配置是不需要的。

这个模型在 hf 那边没有，所以也不存在与其字段兼容的配置了。

File: paddlenlp/transformers/gau_alpha/tokenizer.py [hunk 2]
```diff
@@ -285,3 +287,8 @@ def get_special_tokens_mask(self, token_ids_0, token_ids_1=None, already_has_spe
         if token_ids_1 is not None:
             return [1] + ([0] * len(token_ids_0)) + [1] + ([0] * len(token_ids_1)) + [1]
         return [1] + ([0] * len(token_ids_0)) + [1]
+
+    def get_vocab(self):
+        vocab = {self.convert_ids_to_tokens(i): i for i in range(self.vocab_size)}
+        vocab.update(self.added_tokens_encoder)
+        return vocab
```
Comment (wj-Mcat) [inline paddlenlp/transformers/gau_alpha/tokenizer.py]: 推荐使用这种写法，性能效率比较高：

```suggestion
    def get_vocab(self):
        return dict(self.vocab.token_to_idx, **self.added_tokens_encoder)
```

File: paddlenlp/transformers/gau_alpha/configuration.py (hunk 1 above)
Comment (w5688414) [reply paddlenlp/transformers/gau_alpha/configuration.py]: 已修改

File: paddlenlp/transformers/gau_alpha/modeling.py [hunk 3]
```diff
@@ -243,40 +222,22 @@ class GAUAlphaPretrainedModel(PretrainedModel):
...
         if isinstance(layer, (nn.Linear, nn.Embedding)):
             # In the dygraph mode, use the `set_value` to reset the parameter directly,
             # and reset the `state_dict` to update parameter in static mode.
             if isinstance(layer.weight, paddle.Tensor):
                 num_hidden_layers = (
-                    self.num_hidden_layers
+                    self.config.num_hidden_layers
                     if hasattr(self, "num_hidden_layers")
                     else self.gau_alpha.config["num_hidden_layers"]
                 )
```
Comment (sijunhe) [inline paddlenlp/transformers/gau_alpha/modeling.py]: 这里直接self.config.num_hidden_layers就好了

**Review by sijunhe — COMMENTED** [review]
good PR. 一个小问题改了就可以merge了

File: paddlenlp/transformers/gau_alpha/modeling.py (hunk 3 above)
Comment (w5688414) [reply paddlenlp/transformers/gau_alpha/modeling.py]: 已修改

**Review by sijunhe — APPROVED** [review]
great job!
